src/axom/core/tests/CMakeLists.txt

```diff
@@ -4,19 +4,25 @@
 # SPDX-License-Identifier: (BSD-3-Clause)
 #------------------------------------------------------------------------------
 # Tests for Axom's Core component
+#
+# Note: The tests in this file are consolidated into a few executables.
+# In addition to adding files to the lists below, you'll also have  to #include 
+# your headers in the appropriate 'main' cpp file, e.g. `core_serial_main.cpp`
 #------------------------------------------------------------------------------
 
 #------------------------------------------------------------------------------
-# Specify list of tests
+# Specify list of serial tests
 #------------------------------------------------------------------------------
 
-set(gtest_utils_tests
+set(core_serial_tests
     core_array.hpp
+    core_array_for_all.hpp
     core_execution_for_all.hpp
     core_execution_space.hpp
     core_memory_management.hpp
     core_Path.hpp
     core_stack_array.hpp
+    core_map.hpp
```

Comment:
Note: The `core_map` tests were added to the `core_serial_tests.cpp` executable, but missing from the CMake file. As such, they were being compiled, but not run as part of our unit tests.

After I added this line, it turned out that `core_map` was broken in our Windows/MSVC build. I found and fixed the following two problems:

1. MSVC was not applying the Named Return Value Optimization (nrvo) to `axom::Map`, causing a segfault when it attempted to delete memory twice. I added a move constructor and assignment operator to `Map` as well as some tests to ensure that we can successfully return a Map from a function (as well as supporting `Node`, `Pair` and `Bucket` classes).
2. MSVC was using a different function for `std::hash<int>` than our other compilers.  Our test relied on the fact that the latter were using the identity hash, which implies that if you have `N` integer slots, you can always insert all keys from `0` to `N-1` without rehashing.  This was not the case with MSVC's default hash function.  I refactored the core_map tests to use a custom map with the identify hasher.  I think we'll want to generalize these tests in the future.